core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -4003,6 +4004,33 @@ public static List mapValues(Map map) {
     return new ArrayList<>(map.values());
   }
 
+  /** Support the MAP_FROM_ARRAYS function. */
+  public static Map mapFromArrays(List keysArray, List valuesArray) {
+    if (keysArray.size() != valuesArray.size()) {
```

Comment:
the parameter keysArray and values Array has confirmed is Not NULL?

Reply:
@NobiGo yeap. the calcite supports NullPolicy(6 policy), which will insert code in runtime, spark  using NullIntolerant(only one policy), which is equal to NullPolicy.

for example：
if the NullPolicy.ANY, the calcite will return null  If any of the arguments are null. so the user code doesn't need check whether the input is null

```
defineMethod(MAP_FROM_ARRAYS, BuiltInMethod.MAP_FROM_ARRAYS.method, NullPolicy.ANY);
```